newrelic/terraform-provider-newrelic
# fix(nrql_alert_condition): do not show diff for streaming methods when not provided and defaults are used

# Description

Fixes #1900

When the `aggregation_method`, `aggregation_delay` and `aggregation_timer` fields are not provided in Terraform config it uses default values. This currently results in `terraform plan` diff, even though the values are not changing.

Changes:
* `sinceValueExists` currently evaluates as empty sting with `Get()` (and hence it is never `nil`) (added in [#1620](https://github.com/newrelic/terraform-provider-newrelic/pull/1620)). Switched to `GetOk()`. From [ResourceData.Get](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema#ResourceData.Get):
>Get returns the data for the given key, or nil if the key doesn't exist in the schema. If the key does exist in the schema but doesn't exist in the configuration, then the default value for that type will be returned. For strings, this is "", for numbers it is 0, etc. If you want to test if something is set at all in the configuration, use GetOk.

* `aggregation_method` value comparison is case insensitive now

## Type of change


## Checklist:

## How to test this change?

A basic test case would be (see https://github.com/newrelic/terraform-provider-newrelic/issues/1900 for example):

- terraform apply `newrelic_nrql_alert_condition` without any `aggregation_*`  and `evaluation_offset` properties specified
- `terrafrom plan` => should show no diffs
